include/particle.tcc

```diff
@@ -161,6 +192,18 @@ mpm::HDF5Particle mpm::Particle<Tdim>::hdf5() const {
 
   particle_data.cell_id = this->cell_id();
 
+  particle_data.material_id = this->material_id();
+
+  // Write state variables
+  if (material_ != nullptr) {
+    particle_data.nstate_vars = state_variables_.size();
```

Comment:
I think it's good to throw an error if the `state_variables_.size()>20`. Otherwise, it will cause a segmentation fault while accessing `particle_data.svars[i]`, if `i>19`.

Reply:
Sure!